core/src/main/java/org/sonarsource/sonarlint/core/repository/config/ConfigurationRepository.java

```diff
@@ -119,55 +119,43 @@ public ConfigurationScope getConfigurationScope(String configScopeId) {
+  private Collection<BoundScope> getAllBoundScopes() {
+    return configScopePerId.keySet()
+      .stream()
+      .map(scopeId -> {
+        var effectiveBinding = getEffectiveBinding(scopeId);
+        return effectiveBinding.map(binding -> new BoundScope(scopeId, requireNonNull(binding.getConnectionId()),
+          requireNonNull(binding.getSonarProjectKey()))).orElse(null);
+      })
+      .filter(Objects::nonNull)
       .collect(Collectors.toList());
   }
 
-  public Collection<BoundScope> getBoundScopesByConnection(String connectionId) {
-    return bindingPerConfigScopeId.entrySet().stream()
-      .filter(e -> e.getValue().isBoundToConnection(connectionId))
-      .map(e -> new BoundScope(e.getKey(), connectionId, requireNonNull(e.getValue().getSonarProjectKey())))
-      .collect(Collectors.toList());
+  @CheckForNull
+  public BoundScope getBoundScope(String configScopeId) {
+    var effectiveBinding = getEffectiveBinding(configScopeId);
+    return effectiveBinding.map(binding -> new BoundScope(configScopeId, requireNonNull(binding.getConnectionId()),
+      requireNonNull(binding.getSonarProjectKey()))).orElse(null);
   }
 
-  public List<ConfigurationScope> getConfigScopesWithBindingConfiguredTo(String connectionId) {
-    return bindingPerConfigScopeId.entrySet().stream()
-      .filter(e -> e.getValue().isBoundToConnection(connectionId))
-      .map(e -> configScopePerId.get(e.getKey()))
+  public Collection<BoundScope> getBoundScopesToConnectionAndSonarProject(String connectionId, String projectKey) {
+    return getBoundScopesToConnection(connectionId)
+      .stream()
+      .filter(b -> projectKey.equals(b.getSonarProjectKey()))
       .collect(Collectors.toList());
   }
 
-  public List<BoundScope> getBoundScopesByProject(String projectKey) {
-    return bindingPerConfigScopeId.entrySet().stream()
-      .filter(e -> e.getValue().isBoundToProject(projectKey))
-      .map(e -> new BoundScope(e.getKey(), requireNonNull(e.getValue().getConnectionId()), projectKey))
+  public Collection<BoundScope> getBoundScopesToConnection(String connectionId) {
+    return getAllBoundScopes()
+      .stream()
+      .filter(b -> connectionId.equals(b.getConnectionId()))
       .collect(Collectors.toList());
   }
 
-  public Map<String, Map<String, Set<String>>> getScopeIdsPerProjectKeyPerConnectionId() {
-    Map<String, Map<String, Set<String>>> scopeIdsPerProjectKeyPerConnectionId = new HashMap<>();
-    bindingPerConfigScopeId
-      .forEach((scopeId, bindingConfiguration) -> {
-        if (bindingConfiguration.isBound()) {
-          scopeIdsPerProjectKeyPerConnectionId.computeIfAbsent(bindingConfiguration.getConnectionId(), k -> new HashMap<>())
-            .computeIfAbsent(bindingConfiguration.getSonarProjectKey(), k -> new HashSet<>())
-            .add(scopeId);
-        }
-      });
-    return scopeIdsPerProjectKeyPerConnectionId;
+  public Map<String, Map<String, Set<BoundScope>>> getBoundScopeByConnectionAndSonarProject() {
```

Comment:
This is a mistake, good catch. I added the equals/hashcode in another branch, but I forgot to cherry-pick that. But now I am thinking of it, it might not be the right thing. I will change to a Collection.